### PR TITLE
Use full URLs for favicons

### LINKED ISSUE #247: Favicons should be sourced from canonical url https://transdimension.uk (staging until we flip dns)
- Check they are working once live

See also #210 for complete set needs adding

note if we do #268 withconfigurable env vars for case url, this shoud make use of that

## Acceptance criteria

- [ ] Favicons are sourced from the canonical URL of the current environment (https://transdimension.uk on production)

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Closes #247

## Description

Using environment constant `canonicalUrl`

### CODE REVIEW

File: src/Site.elm [hunk 1]
```diff
@@ -71,6 +72,11 @@ pathFromString srcString =
     Pages.Url.fromPath <| Path.fromString srcString
 
 
+fullPathFromString : String -> Pages.Url.Url
+fullPathFromString srcString =
+    Pages.Url.external (String.join "" [ chompTrailingUrlSlash canonicalUrl, srcString ])
```
Comment (aaaaargZombies) [inline src/Site.elm:77]:
```suggestion
    Pages.Url.external <| chompTrailingUrlSlash canonicalUrl ++ srcString
```
Maybe this is slightly better string smooshing, but it achieves the same thing.

**Review by aaaaargZombies — DISMISSED** [review]
This works on my machine, just a small note that I will leave up to you.

File: src/Site.elm [hunk 2]
```diff
@@ -33,7 +34,7 @@ data =
 head : Data -> List Head.Tag
 head static =
     [ Head.sitemapLink "/sitemap.xml"
-    , Head.appleTouchIcon (Just 180) (pathFromString "/favicons/apple-touch-icon.png")
+    , Head.appleTouchIcon (Just 180) (fullPathFromString "/favicons/apple-touch-icon.png")
```
Comment (aaaaargZombies) [inline src/Site.elm:37]: Did this used to produce a relative url?
Is full domain equivalent to and external url?

Comment (aaaaargZombies) [reply src/Site.elm:37]: ah ok, I started adding an `AbsoluteInternal` type to account for this situation then I realized I was in the library code not our own so I shouldn't meddle with it. A bit strange that elm-pages makes that decision for you up front about how to format internal URLs

**Comment by aaaaargZombies** [comment]
OK I'm fairly certain that the `main` branch is producing absolute URLS combining the `Constant.canonicalUrl` + `Internal` URL type. What is a bit confusing is that if you change the `canonicalUrl` in the `.env` file and run `npm start` the dev server still runs on `localhoust:3030` but if you inspect the page in the browser you can see the change. If you change it run `npm run build` then just serve it yourself you can see it all working. 

so perhaps we should close this PR and mark the issue as OK?

File: src/Site.elm [hunk 3]
```diff
@@ -71,6 +72,11 @@ pathFromString srcString =
     Pages.Url.fromPath <| Path.fromString srcString
```
[the remark this answers is not in the document]
Comment (aaaaargZombies) [reply src/Site.elm:72]: In the `icons` function below there is still a reference to this function but presumably if we need the new function for one we need it for all of them.

File: src/Site.elm (hunk 1 above)
Comment (aaaaargZombies) [inline src/Site.elm:77]:
```suggestion
absoluteUrl : String -> Pages.Url.Url
absoluteUrl srcString =
    srcString
        |> Path.fromString
        |> Pages.Url.fromPath
        |> Pages.Url.toAbsoluteUrl canonicalUrl
        |> Pages.Url.external
```
I also came up with this which made sense based on the package types and the functions but it feels like its going round the houses a bit just to join two strings.

**Comment by ivan-kocienski-gfsc** [comment]
This fix is no longer required as the ticket was closed for other reasons

**Pull request closed without merging**